u2f-gae-demo/src/com/google/u2f/gaedemo/storage/TokenStorageData.java

```diff
@@ -118,10 +118,12 @@ public boolean equals(Object obj) {
         && Arrays.equals(this.attestationCert, that.attestationCert);
   }
 
-  private static X509Certificate parseCertificate(byte[] encodedDerCertificate) {
+  private static X509Certificate[] parseCertificateChain(byte[] encodedDerCertificates) {
     try {
-      return (X509Certificate) CertificateFactory.getInstance("X.509").generateCertificate(
-          new ByteArrayInputStream(encodedDerCertificate));
+      Collection<? extends Certificate> certCollection =
```

Comment:
Comment from you: this expression should be replaced with a call to a method in X509Util.